lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -288,6 +289,115 @@ describe('ContentMetadataComponent', () => {
             expect(contentMetadataService.getGroupedProperties).toHaveBeenCalled();
         }));
 
+        describe('Save button - Grouped Properties', () => {
+            beforeEach(() => {
+                getGroupedPropertiesSpy.and.returnValue(
+                    of([
+                        {
+                            editable: true,
+                            title: 'test',
+                            properties: []
+                        }
+                    ])
+                );
+
+                component.ngOnInit();
+                component.readOnly = false;
+            });
+
+            it('should disable the save button if metadata has not changed', () => {
+                component.hasMetadataChanged = false;
+                fixture.detectChanges();
+                toggleEditModeForGroup();
+                expect(getGroupSaveButton().disabled).toBeTrue();
+            });
+
+            it('should disable the save button if there are invalid properties', () => {
+                component.hasMetadataChanged = true;
+                component.invalidProperties = ['invalidPropertyKey'];
```

Comment:
I won't add this comment to every place but please apply it for each test for invalidProperties. Generally invalidProperties is normal property, it's not Input property. So you shouldn't test it by just setting this field because normally that field won't be set from parent. Instead it will be set inside ngOnInit code. So to test that you should to run change detector or ngOnInit function to set that property. 
Please also change the same for hasMetadataChanged property because I see it's not Input as well.

Reply:
done